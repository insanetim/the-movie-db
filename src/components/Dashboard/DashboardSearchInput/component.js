import PropTypes from 'prop-types'
import { Row, Col, Input } from 'antd'

import useContainer from './hook'

const DashboardSearchInput = ({ searchQuery }) => {
  const { currentValue, handleChange, handleSearch } = useContainer(searchQuery)

  return (
    <Row
      justify='center'
      gutter={{
        xs: 8,
        sm: 16,
        md: 24,
        lg: 22
      }}
    >
      <Col
        xs={{ span: 20 }}
        sm={{ span: 20 }}
        md={{ span: 14 }}
        lg={{ span: 12 }}
        xl={{ span: 10 }}
      >
        <Input.Search
          className='top-margin'
          placeholder='Enter movie name'
          size='large'
          enterButton='Search'
          allowClear
          value={currentValue}
          onChange={handleChange}
          onSearch={handleSearch}
        />
      </Col>
    </Row>
  )
}

DashboardSearchInput.propTypes = {
  searchQuery: PropTypes.string
}

DashboardSearchInput.defaultProps = {
  searchQuery: null
}

export default DashboardSearchInput
