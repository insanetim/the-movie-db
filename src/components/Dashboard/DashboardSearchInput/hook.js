import { useState, useEffect } from 'react'
import { useDispatch } from 'react-redux'
import { useNavigate } from 'react-router-dom'
import { clearSearch } from 'src/state/dashboard/actions'

const useContainer = searchQuery => {
  const dispatch = useDispatch()
  const [currentValue, setCurrentValue] = useState(searchQuery)
  const navigate = useNavigate()

  const handleChange = event => {
    setCurrentValue(event.target.value)
  }

  const handleSearch = value => {
    if (value.trim()) {
      navigate({ pathname: '/', search: `search=${value}` })
    } else {
      dispatch(clearSearch())
      navigate('/')
    }
  }

  useEffect(() => {
    setCurrentValue(searchQuery)
  }, [searchQuery])

  return { currentValue, handleChange, handleSearch }
}

export default useContainer
