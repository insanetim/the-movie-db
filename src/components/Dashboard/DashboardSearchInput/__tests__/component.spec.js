import { shallow } from 'enzyme'

import DashboardSearchInput from '../component'

const mockedHookData = {
  currentValue: 'test/searchQuery',
  handleChange: jest.fn(),
  handleSearch: jest.fn()
}
jest.mock('../hook', () => jest.fn(() => mockedHookData))

describe('CreateListModal component tests', () => {
  const component = shallow(<DashboardSearchInput />)

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
