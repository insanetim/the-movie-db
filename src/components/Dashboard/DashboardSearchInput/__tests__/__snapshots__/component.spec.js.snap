// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateListModal component tests matches snapshot 1`] = `
<Row
  gutter={
    {
      "lg": 22,
      "md": 24,
      "sm": 16,
      "xs": 8,
    }
  }
  justify="center"
>
  <Col
    lg={
      {
        "span": 12,
      }
    }
    md={
      {
        "span": 14,
      }
    }
    sm={
      {
        "span": 20,
      }
    }
    xl={
      {
        "span": 10,
      }
    }
    xs={
      {
        "span": 20,
      }
    }
  >
    <Search
      allowClear={true}
      className="top-margin"
      enterButton="Search"
      onChange={[MockFunction]}
      onSearch={[MockFunction]}
      placeholder="Enter movie name"
      size="large"
      value="test/searchQuery"
    />
  </Col>
</Row>
`;
